cmake/CorrosionGenerator.cmake

```diff
@@ -134,16 +145,18 @@ function(_generator_add_package_targets)
             # name consistent independent of the Rust version. `bin` target names are not affected.
             # See https://github.com/corrosion-rs/corrosion/issues/501 for more details.
             string(REPLACE "\-" "_" target_name "${target_name}")
+            string(REPLACE "\-" "_" target_name_cmake "${target_name_cmake}")
 
             set(archive_byproducts "")
             set(shared_lib_byproduct "")
             set(pdb_byproduct "")
 
-            add_library(${target_name} INTERFACE)
-            _corrosion_initialize_properties(${target_name})
+            add_library(${target_name_cmake} INTERFACE IMPORTED GLOBAL)
```

Comment:
Whats the reason to make the interface library `IMPORTED` and `GLOBAL`? It doesn't seem directly related to this PR.